### PR TITLE
Suspended Users Cannot Purchase Tickets

### PR DESCRIPTION
**Pull request by cheljoh**
Users with suspended accounts cannot create a new order.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -11,6 +11,10 @@ def current_user
     @user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
+  def active_user?
+    current_user.status == "active"
+  end
+
   def current_admin?
     current_user && current_user.admin?
   end
```
Comment (damwhit) [inline app/controllers/application_controller.rb]: Maybe try active record query?

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -24,8 +24,16 @@ class OrdersController < ApplicationController
...
+      @tickets = OrderProcessor.new(@cart).tickets
+      @order = Order.new
+      render :new
+    else
+      flash[:info] = "sorry, you cannot purchase tickets :("
+      @cart.empty_cart
+      session[:cart] = @cart.contents
+      redirect_to root_path
+    end
   end
```
Comment (damwhit) [inline app/controllers/orders_controller.rb:39]: Don't let those suspended users add stoof!  - I like it.

**Comment by damwhit** [comment]
@cheljoh looks good to me!  @erinnachen wanna take a look see?

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -24,8 +24,16 @@ class OrdersController < ApplicationController
...
   def new
-    @tickets = OrderProcessor.new(@cart).tickets
-    @order = Order.new
+    if active_user?
+      @tickets = OrderProcessor.new(@cart).tickets
+      @order = Order.new
+      render :new
+    else
+      flash[:info] = "sorry, you cannot purchase tickets :("
+      @cart.empty_cart
```
Comment (erinnachen) [inline app/controllers/orders_controller.rb:33]: I like the cart empty.

**Comment by erinnachen** [comment]
Those suspended users.... grrrrrr.

**Pull request merged**